### DISCUSSION TITLE
Can anyone suggest me a image dataset for binary classification of weather the image has object in it or not?

### DISCUSSION THREAD

**Discussion by sanchyan** [mrdbourke/pytorch-deep-learning#173] [Q&A]
Hello everyone I am going through the youtube video course and I want to write a binary classification model for detecting if an image has something in it. 
For example, if there is a dog in an image, it must be labeled as not_empty; if the image is just a scene with no animals in it, then the image must be labeled as empty.

My questions:

-  Do anybody know such kind of dataset where I can try empty or not-empty binary classification problem?
-  What can be the ways to find if an image is empty or not empty?

Thank You 
I hope to get some good suggestions. Love you all.

**Comment by ghost** [comment]
Maybe this one!  Face mask detection dataset from Kaggle

https://www.kaggle.com/datasets/ashishjangra27/face-mask-12k-images-dataset

**Comment by mrdbourke** [comment]
If you can't find a dataset, you could start by creating your own (e.g. start with ~25-50 images of each class), then use the model itself to go over different images and continually grow a dataset of new images.

This image dataset was released recently with/without subjects - https://arxiv.org/abs/2201.03101 (I think it is mostly focused on people though rather than non-people).

**Reply by sanchyan** [reply]
Thank you, brother. I am preparing my own dataset with the images from the wildlife dataset available on the internet from different wildlife camera trap datasets.
Unfortunately, I have some more questions as I am new to this field.

I tried to fit my data as in your custom dataset video. Now I have collected around 1031empty images (with no animal in them) and 1224 non-empty images (with any animal in them).

![image](https://user-images.githubusercontent.com/43085459/202150710-385286ad-58a9-4991-9dae-fab5048fe2cb.png)


I made several models with different data transforms. my best one now is with center crop and normalization with mean and std of the dataset. I have used the TinyVGG architecture you have used and will see the transfer learning section from your book version. since the camera trap images are even complex for my eye to find an animal in the picture especially if it is a rat or squirrel that has triggered the camera trap.

Now the question is:

-     **I have chosen the image classification method (Is this the best way?) or should I use the object detection method with the bounding box? (To do so for this dataset it will take eternity just to label the images).  Can you give some suggestions on this?**

![image](https://user-images.githubusercontent.com/43085459/202151331-774fc0d6-22f4-4e2e-9e7f-68c75d473dca.png)
